test/NCalc.Tests/CustomCultureTests.cs

```diff
@@ -65,36 +65,23 @@ void ExecuteTest(string formula, object expectedValue)
             }.Evaluate());
         }
     }
-    
+
     [Fact]
     public void ShouldParseInvariantCulture()
     {
-        var originalCulture = (CultureInfo)Thread.CurrentThread.CurrentCulture.Clone();
-        try
-        {
-            var culture = (CultureInfo)CultureInfo.InvariantCulture.Clone();
-            culture.NumberFormat.NumberDecimalSeparator = ",";
-            Thread.CurrentThread.CurrentCulture = culture;
-            var exceptionThrown = false;
-            try
-            {
-                var expr = new Expression("[a]<2.0") { Parameters = { ["a"] = "1.7" } };
-                expr.Evaluate();
-            }
-            catch (FormatException)
-            {
-                exceptionThrown = true;
-            }
+        var e = new Expression("[a]<2.0", CultureInfo.InvariantCulture);
+        e.Parameters["a"] = "1.7";
+        Assert.Equal(true, e.Evaluate());
+    }
 
-            Assert.True(exceptionThrown);
+    [Fact]
+    public void ShouldParseCultureWithCommaSeparator()
+    {
+        var culture = (CultureInfo)CultureInfo.InvariantCulture.Clone();
+        culture.NumberFormat.NumberDecimalSeparator = ",";
 
-            var e = new Expression("[a]<2.0", CultureInfo.InvariantCulture);
-            e.Parameters["a"] = "1.7";
-            Assert.Equal(true, e.Evaluate());
-        }
-        finally
-        {
-            Thread.CurrentThread.CurrentCulture = originalCulture;
-        }
+        var e = new Expression("[a]<2,0", culture);
```

Comment:
You will need to add something like `Round(3,123)` and see how it behaves.
I don't think allowing any decimal separator anywhere is sustainable.

Reply:
```
var culture = (CultureInfo)CultureInfo.InvariantCulture.Clone();
culture.NumberFormat.NumberDecimalSeparator = ",";

var e = new Expression("Round(3,123,0)", culture);
Assert.Equal(3d, e.Evaluate());
```

Test is passing

Reply:
I agree, it's not looking good, it'll be better to replace NCalc parameter delimiter to semicolon, instead of comma

Reply:
> Assert.Equal(3d, e.Evaluate());

Is it `0` because it just discarded the "second" argument? What about `If([Pi]>3.14,3.14,3.13)` but with commas ;)

Reply:
> > Assert.Equal(3d, e.Evaluate());
> 
> Is it `0` because it just discarded the "second" argument? What about `If([Pi]>3.14,3.14,3.13)` but with commas ;)

```
var culture = (CultureInfo)CultureInfo.InvariantCulture.Clone();
culture.NumberFormat.NumberDecimalSeparator = ",";

var e = new Expression("if([Pi] > 3,14, 3,14, 3,13)", culture);
e.Parameters["Pi"] = Math.PI;
Assert.Equal(3.14d, e.Evaluate());
```
Test is passing

Reply:
Cheater ;) 

I assume that could work if you just passed the decimal separator and an argument separator which are distinct, like `,` and `;`. As long as the user decides, their problem.

I personally don't care, just explaining all the issues I can see.

Reply:
I love your idea to make it possible to use `,` and `;` as argument separator. Thank you for your inputs!